scanapp/tasks.py

```diff
@@ -74,6 +80,55 @@ def scan_code_async(URL, scan_id, path):
         folder_name = None
         apply_scan_async.delay(path, scan_id, scan_type, URL, folder_name)
 
+
+@app.task
+def scan_code_async_final(URL, scan_id, scan_type):
+    """
+    Create and save a file at `path` present at `URL` using `scan_id` and bare `path`
+    and apply the scan.
+    """
+    logger.info('git repo detected')
+
+    dir_name = 'temp'
+
+    home_path = expanduser("~")
+
+    os.chdir(home_path)
+
+    if os.path.isdir(dir_name):
+        shutil.rmtree(dir_name)
+
+    os.mkdir(dir_name)
+
+    repo = git.Repo.init(dir_name)
+    origin = repo.create_remote('origin', URL)
+    origin.fetch()
+    origin.pull(origin.refs[0].remote_head)
+
+    logger.info('Done ! Remote repository cloned')
+
+    filename = home_path + '/temp/'
+
+    # print filename
+
+    path = filename
+
+    folder_name = None
+
+    apply_scan_async.delay(path, scan_id, scan_type, URL, folder_name)
+
+
+@app.task
+def scan_code_async_special(URL, scan_id, path):
+    """
+    Applying scan on the cloned repository
+    """
+    scan_type = 'SpecialUrl'
+    folder_name = path
+
+    apply_scan_async.delay(path, scan_id, scan_type, URL, folder_name)
+
```

Comment:
Except for this line, the body of this function should be in a non-task function that can be tested independently